Dockerfile

```diff
@@ -28,8 +35,10 @@ COPY --from=builder /build/v3 /var/www/html/v3
 
 # Offline build: cache .theia and .vsix files in registry itself and update metas
 FROM builder AS offline-builder
-RUN ./cache_artifacts.sh v3
-RUN chmod -R g+rwX /build
+
+# To only cache files from /latest/ folders, use ./cache_artifacts.sh v3 --latest-only 
+# and uncomment line above to remove files so they're not included in index.json -- RUN rm -fr $(find /build/v3 -name 'meta.yaml' | grep -v "/latest/")
```

Comment:
Is this necessary? If an earlier step removes non-latest metas, then `cache_artifacts.sh` won't see the removed files and so only latest will be cached naturally.

Reply:
yes, `remove files so they're not included in index.json` is not the same as choosing to cache or not cache them locally.

This will delete files; the cache_artifacts does not delete stuff, it only caches/creates stuff.

Reply:
But `cache_artifacts` works by reading the extensions from the metas -- if the metas no longer exist it can't read them.

Reply:
Exactly. So we have to delete them if we want cache_artifacts to not see and therefore cache them. It's three steps:

1. delete meta.yaml files we don't want to include in index.json
2. generate index.json (excluding the deleted files!) w/ `index.sh`
3. cache only the /latest/ meta.yaml files' vsix/theia files with `cache_artifacts.sh`

I considered adding the delete functionality to the cache_artifacts script but if we do that we'd have to call the `index.sh` script again so we don't point at meta.yaml files that no longer exist.

Alternatively we could add a -latest-only flag to the index.sh script ... if you prefer that?

Reply:
If there's a use-case for caching only latest plugins but keeping all metas available (so you can choose latest and it's cached, but if you chose any other version it has to download), then a `--latest-only` option could be useful. Honestly, though, that seems too complicated for so little gain (except your image would be smaller, I suppose). My preferred approach would be something like

- We support 2x2 options:
    - You can build offline or not (already implemented for the most part)
    - You can include all versions or only latest. If only latest, no other metas are stored in the registry.

The minimal changes to support this would be to add a way to delete non-latest metas in the builder phase, then everything would follow naturally without additional changes.

Reply:
aaand I already have that.

```
RUN rm -fr $(find /build/v3 -name 'meta.yaml' | grep -v "/latest/")
```

in: 

* Dockerfile line 22 (optional, commented out): https://github.com/eclipse/che-plugin-registry/pull/232/files#diff-3254677a7917c6c01f55212f86c57fbfR22
* rhel.Dockerfile line 75 (optional, commented out): https://github.com/eclipse/che-plugin-registry/pull/232/files#diff-75aafc399136415e2e24dad895058632R75